// @bit/ahojukka5.bloglist.notification/dist/Notification.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Snackbar = _interopRequireDefault(require("@material-ui/core/Snackbar"));

var _IconButton = _interopRequireDefault(require("@material-ui/core/IconButton"));

var _Close = _interopRequireDefault(require("@material-ui/icons/Close"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var useStyles = (0, _styles.makeStyles)(function (theme) {
  return {
    close: {
      padding: theme.spacing(0.5)
    }
  };
});
/**
 * Renders a notification message in bottom of the screen.
 */

var Notification = function Notification(_ref) {
  var message = _ref.message,
      onClose = _ref.onClose;
  var classes = useStyles();
  return _react.default.createElement(_Snackbar.default, {
    anchorOrigin: {
      vertical: 'bottom',
      horizontal: 'left'
    },
    open: message ? true : false,
    autoHideDuration: 6000,
    onClose: onClose,
    ContentProps: {
      'aria-describedby': 'message-id'
    },
    message: _react.default.createElement("span", {
      id: "message-id"
    }, message),
    action: [_react.default.createElement(_IconButton.default, {
      key: "close",
      "aria-label": "close",
      color: "inherit",
      className: classes.close,
      onClick: onClose
    }, _react.default.createElement(_Close.default, null))]
  });
};

Notification.propTypes = {
  /** Message to show. */
  message: _propTypes.default.string,

  /** Function to call when notification is closed. */
  onClose: _propTypes.default.func
};
var _default = Notification;
exports.default = _default;
